bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -1768,7 +1768,7 @@ private static void execBinaryOpCodes(WorkerExecutionContext ctx, WorkerData sf,
                 i = operands[0];
                 j = operands[1];
                 k = operands[2];
-                sf.intRegs[k] = sf.refRegs[i] != sf.refRegs[j] ? 1 : 0;
+                sf.intRegs[k] = !isReferenceEqual(sf.refRegs[i], sf.refRegs[j]) ? 1 : 0;
```

Comment:
This works, but not optimal because it doesn't fail fast. Ideally, we should write separate inverse logic.